hal/hal_uv_client_usb.c

```diff
@@ -29,6 +29,10 @@ typedef struct adb_client_usb_s {
     /* FIXME libuv handle must be right after adb_client_uv_t */
     uv_pipe_t read_pipe;
     uv_pipe_t write_pipe;
+#ifdef CONFIG_ADBD_USB_HOTPLUG_BYTIMER
+    uv_timer_t timer;
+#endif
+    char path[0];
```

Comment:
I would prefer to handle the hotplug timer outside of the hal_uv_client_usb file. This new hotplug timer could periodically check for "/dev/adb0" file, and call `adb_uv_usb_setup(adbd, "/dev/adb0")` when required to setup usb port.